packages/buidler-core/src/internal/buidler-evm/provider/modules/eth.ts

```diff
@@ -464,26 +482,72 @@ export class EthModule {
 
   private async _getFilterChangesAction(
     filterId: BN
-  ): Promise<string[] | null> {
-    const id = filterId.toNumber(); // This may throw, but it's ok
+  ): Promise<string[] | RpcLogOutput[] | null> {
+    const id = filterId.toNumber();
```

Comment:
Now that I see it, using number here is somewhat inconsistent with my other comments about block numbers. The only difference I see is that filter ids are only used here and in `Node`, while block numbers are mostly an ethereumjs thing. I'm not really comfortable with this cast either to be honest, and I guess I was a little lazy when coding this.

Reply:
Initially, I implemented filters as `map<string, Filter>`, but after I saw `toNumber()` in `newBlockFilter` I switched to `map<number, Filter>`. Makes sense to use string as key.